src/assign.c

```diff
@@ -744,6 +744,12 @@ SEXP assign(SEXP dt, SEXP rows, SEXP cols, SEXP newcolnames, SEXP values)
 #define MSGSIZE 1000
 static char memrecycle_message[MSGSIZE+1]; // returned to rbindlist so it can prefix with which one of the list of data.table-like objects
 
+static char column_desc[MSGSIZE+1]; // can contain column names, hence relatively large allocation.
+const char *columnDesc(int colnum, const char *colname) {
+  snprintf(column_desc, MSGSIZE, _("column %d named '%s'"), colnum, colname);
```

Comment:
Can we add a comment [to be extracted by `xgettext -cTRANSLATORS`](https://www.gnu.org/software/gettext/manual/html_node/xgettext-Invocation.html#index-_002dc_002c-xgettext-option) saying

```c
/* TRANSLATORS: use nominative case; string will be used inside parentheses */
```

Reply:
WDYT about using ` %s` instead of ` (%s)` in the longer string, then `columnDesc` returns `(column %d names '%s')`, then the parenthetical won't need the comment?

Reply:
Makes perfect sense, this should be even better!